### PR TITLE
Install Stave via pip

### LINKED ISSUE #183: Install Stave via pip
**Is your feature request related to a problem? Please describe.**
Currently users need to manually install `django`, `react.js` and other dependencies
 to boot a Stave instance. Is there any way we can wrap it to a standalone package that can be `pip` installed.

**Describe the solution you'd like**
Package the Stave project with required dependencies according to [this tutorial](https://packaging.python.org/tutorials/packaging-projects/).

**Describe alternatives you've considered**
May need to include the `build` folder for `react.js` rendering and `db.sqlite3` for `django` backend.

**Additional context**
Should support viewer mode and full mode.

### PR DESCRIPTION
**Pull request by mylibrar**
This PR fixes #183. This PR is a follow up to #184.

### Description of changes
- Add **`setup.py`** and **`MANIFEST.in`** to package Stave. `MANIFEST.in` includes two paths, i.e., `sample_sql/` and `build/`, which are not included in this PR but will be generated when running Github workflow **`pypi.yml`**. **`package-lock.json`** is included to ensure that workflow can work. The workflow will run the following steps to construct a package structure:
    - Install `npm` and generate `build` folder, which will then be moved to the package path `stave/simple-backend/nlpviewer_backend/lib/build`.
    - Copy `sample_sql/` to `stave/simple-backend/nlpviewer_backend/sample_sql/`.
    - **`settings.py`** is modified to reroute database to the new path and to hide `SECRET_KEY`. The updated `settings.py` is stored under `scripts` and will be moved to package path during the workflow.
- Update `StaveViewer` interface: remove the `build_path` parameter since it is already copied to the same directory of **`stave_viewer.py`**. **`manageproject.py`** is also updated to reflect this change.
- Add `load_database()` to `StaveViewer` to support initializing database `db.sqlite3` and loading data from `sample_sql/*.sql`.
- Fix issues in **`stave_project.py`** to allow overwriting textpacks with the same index.

### Demonstration of results
- Install
    The source archive and built distribution are already uploaded to [TestPyPI](https://test.pypi.org/project/stave/0.0.1.dev0/). Run the following command to install:
        
    > pip3 install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple stave==0.0.1.dev0

- Test Snippet
    > from nlpviewer_backend.lib.stave_viewer import StaveViewer
sv = StaveViewer(in_viewer_mode=False)
sv.run()
sv.load_database(load_auth=True, load_samples=True) #Optional
sv.open()

### CODE REVIEW

**Comment by mylibrar** [comment]
Stave CLI (`entry_points.console_scripts`) will be separated to another PR for better tracking.

File: .github/workflows/pypi.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ env.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.python-version }}    
+      - name: Install pypa/build
+        run: |
+          python -m pip install build --user
+      - name: Check current directory
+        run: |
```
Comment (hunterhector) [inline .github/workflows/pypi.yml:24]: this `ls` is used for debug only, so if you don't need it you can remove it.

File: .github/workflows/pypi.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          npm install
+          npm run build
+          mv ./build ./simple-backend/nlpviewer_backend/lib/build
+      - name: Copy sql statements to package path
+        run: |
+          cp -r ./simple-backend/sample_sql ./simple-backend/nlpviewer_backend/sample_sql
+      - name: Replace the repo version of settings.py with package version
+        run: |
+          mv ./scripts/settings.py ./simple-backend/nlpviewer_backend/settings.py
```
Comment (hunterhector) [inline .github/workflows/pypi.yml:41]: Can you explain why we need to do this? We can also add comments here to make future developers understand this.

File: .github/workflows/pypi.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+           pwd
+           ls
+      - name: Set up node.js 14
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - name: Build react.js application and move to package path
+        run: |
+          npm install
+          npm run build
```
Comment (hunterhector) [inline .github/workflows/pypi.yml:34]: We are using `yarn` to build packages directly and we have `yarn.lock` in our source. Maybe we should stick with Yarn for this PR.

File: scripts/settings.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
+"""
+Django settings for nlpviewer_backend project.
+
+Generated by 'django-admin startproject' using Django 2.1.7.
```
Comment (hunterhector) [inline scripts/settings.py:4]: If this file is generated, should we commit it?

**Review by hunterhector — COMMENTED** [review]
We are currently using `yarn.lock` to maintain the dependency tree. According to this question:
https://stackoverflow.com/questions/44552348/should-i-commit-yarn-lock-and-package-lock-json-files

We either only commit one file, or use a process to make the sync. In this PR, let's stick with `yarn` first.

**Comment by mylibrar** [comment]
`scripts/settings.py` and `package-lock.json` are removed. The latest package for test is [stave==0.0.1.dev4](https://test.pypi.org/project/stave/0.0.1.dev4/), which can be installed via
> pip3 install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple stave==0.0.1.dev4

**Review by hunterhector — APPROVED** [review]

**Pull request merged**